pkg/controller/quota/quota.go

```diff
@@ -146,16 +146,25 @@ func addStorage(appInstance *v1.AppInstance, quotaRequest *adminv1.QuotaRequestI
 			size = boundSize
 		}
 
-		// No need to proceed if the size is empty
-		if size == "" || size == "0" {
+		// Handle three cases:
+		// 1. The volume's size is explicitly set to 0. This means there is nothing to count.
+		// 2. The volume's size is not set. This means we should assume the default size.
+		// 3. The volume's size is set to a specific value. This means we should use that value.
+		var sizeQuantity resource.Quantity
+		switch size {
```

Comment:
It looks like the size of bound volumes are counted here. Is there any risk that we are double-counting bound volumes? Or are we saying you can have as many volumes as you want as long as none of them are bound?